torch/csrc/jit/passes/specialize_autogradzero.cpp

```diff
@@ -10,15 +10,49 @@ namespace jit {
 
 struct AutogradZeroSpecializer {
   enum class State { Nonzero, Zero, Unknown };
+
   AutogradZeroSpecializer(std::shared_ptr<Graph> graph)
       : graph_(std::move(graph)) {}
 
   void run() {
-    setStatesOnGraphInputs();
-    specializeAutogradOps(graph_->block());
+    if (!isBackwardGraph()) {
+      return;
+    }
+    if (getProfilingMode()) {
+      if (auto vif = guardSpecializations()) {
```

Comment:
Does `vif` stand for a "versioning if"? A comment could be helpful here I think.